reports/helm-releases/main.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/reports/pkg/hoodaw"
+)
+
+var (
+	bucket         = flag.String("bucket", os.Getenv("KUBECONFIG_S3_BUCKET"), "AWS S3 bucket for kubeconfig")
+	ctxLive        = flag.String("contextLive", "live.cloud-platform.service.justice.gov.uk", "Kubernetes context specified in kubeconfig")
+	ctxManager     = flag.String("contextManager", "manager.cloud-platform.service.justice.gov.uk", "Kubernetes context specified in kubeconfig")
+	ctxLive_1      = flag.String("contextLive_1", "live-1.cloud-platform.service.justice.gov.uk", "Kubernetes context specified in kubeconfig")
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawEndpoint = flag.String("hoodawEndpoint", "/helm_whatup", "Endpoint to send the data to")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	kubeconfig     = flag.String("kubeconfig", "kubeconfig", "Name of kubeconfig file in S3 bucket")
+	region         = flag.String("region", os.Getenv("AWS_REGION"), "AWS Region")
+
+	endPoint = *hoodawHost + *hoodawEndpoint
+)
+
+type helmNamespace struct {
+	Namespace string
+}
+
+type helmRelease struct {
+	Name             string `json:"name"`
+	Namespace        string `json:"namespace"`
+	InstalledVersion string `json:"installed_version"`
+	LatestVersion    string `json:"latest_version"`
+	Chart            string `json:"chart"`
+}
+
+type resourceMap map[string]interface{}
+
+func main() {
+
+	// export kube config file path.
+	os.Setenv("KUBECONFIG", "/tmp/config")
+
+	helmReleaseLive, err := getHelmReleases("live")
+	if err != nil {
+		log.Fatalln("error in getting helm releases")
+	}
+
+	helmReleaseManager, err := getHelmReleases("manager")
+	if err != nil {
+		log.Fatalln("error in getting helm releases")
+	}
+
+	helmReleaseLive1, err := getHelmReleases("live-1")
+	if err != nil {
+		log.Fatalln("error in getting helm releases")
+	}
```

Comment:
I feel like there might be a better way of doing this. You could create a slice of contexts:
```
contexts []string{ctxLive, ctxManager, ctxLive1}
```

and then loop over context and then you don't need to switch/case:

```
for _, cluster := range context {
  helmRelease, err := getHelmReleases(cluster)
  if err != nil {
    return err
  }
}

Reply:
Then you have a map of Helm Releases and you can do what you like with it.

Reply:
and maybe even submit it to the API.

Reply:
If you wanted to give the cluster a nicer name you could slice it up.
```
var releases map(string)[]helmReleases
for _, cluster := range context {
  helmRelease, err := getHelmReleases(cluster)
  if err != nil {
    return err
  }

  strings.SplitAfter(cluster, ".")[0][helmRelease]
}

```

Reply:
Done.  loop using context use strings.Split(ctx, ".")[0] instead of SplitAfter as that will split along with the separator